microhapulator/api.py

```diff
@@ -724,14 +725,13 @@ def repetitive_mapping(marker_bam_file, fullref_bam_file, markertsv, minbasequal
     microhaps = MicrohapIndex.from_files(markertsv)
     if not microhaps.has_chrom_offsets:
         raise ValueError("cannot perform repetitive analysis without chromosome offsets")
-    reads_to_marker = get_reads_in_marker_loci(fullref_bam_file, microhaps)
-    for locus in microhaps.loci.values():
+    reads_to_marker_fullref = get_reads_in_marker_loci(fullref_bam_file, microhaps)
+    for locus, marker in microhaps:
         repetitive_count = count_repetitive_reads(
-            marker_bam, locus, reads_to_marker, minbasequal=minbasequal
+            marker_bam, locus, marker, reads_to_marker_fullref, minbasequal=minbasequal
         )
-        for marker in locus:
-            counts["Marker"].append(marker.id)
-            counts["RepetitiveReads"].append(repetitive_count)
+        counts["Marker"].append(marker.id)
+        counts["RepetitiveReads"].append(repetitive_count)
```

Comment:
Previously, the repetitive reads were being counted at the locus level, and then that count was assigned to every marker at that locus, so all markers at a locus have the same value.

Reply:
Makes sense.